winml/lib/Api.Ort/OnnxruntimeEngineBuilder.cpp

```diff
@@ -41,6 +41,10 @@ STDMETHODIMP OnnxruntimeEngineBuilder::CreateEngine(_winml::IEngine** out) {
     RETURN_HR_IF_NOT_OK_MSG(ort_api->AddFreeDimensionOverride(session_options.get(), DATA_BATCH, batch_size_override_.value()),
                             ort_api);
   }
+  for (auto&& override : named_dimension_overrides_) {
+    std::string narrow_name = std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(override.Key().c_str());
+    ort_api->AddFreeDimensionOverrideByName(session_options.get(), narrow_name.c_str(), override.Value());
```

Comment:
Xiang identified dml perf issues with using this api, will you be doing the work here to make sure that these overrides are persisted into dml as well?

Reply:
ORT applies the graph transformer at session creation time so this should be okay. Sheil verified this with Jeff from DML.